demos/tests/cases.py

```diff
@@ -68,6 +71,59 @@ def fixed_args(self, source_dir, build_dir):
             *(['-l', str(cpu_extension_path)] if cpu_extension_path.exists() else [])]
 
 
+class NotebookDemo(Demo):
+    """
+    Converts notebook to a Python script with nbconvert. Adds code to make the
+    script work with command line parameters. The `test_code` parameter is
+    added at the end of the script. This should contain the line of code to
+    execute to test the notebook.
+    """
+
+    def __init__(self, name, implementation='jupyter_python', device_keys=None, test_cases=None, test_code=None):
+        super().__init__(name, implementation, device_keys, test_cases)
+        self._exec_name = self._exec_name.replace('_jupyter_python', '')
+        self.test_code = test_code
+
+    def fixed_args(self, source_dir, build_dir):
+        demo_env = {**os.environ,
+                   'PATH': f"{os.environ['PATH']}{os.pathsep}"
+                           f"{os.path.dirname(sys.executable)}",
+                   'PYTHONPATH': f"{os.environ['PYTHONPATH']}{os.pathsep}"
+                                 f"{os.path.join(os.path.dirname(os.path.abspath(os.curdir)), 'common', 'python')}"}
+
+        notebook_file = str(source_dir / self.subdirectory / (self._exec_name + '.ipynb'))
+        python_file = notebook_file.replace('ipynb', 'py')
+        python_test_file = python_file.replace('.py', '_test.py')
+        subprocess.run(['jupyter', 'nbconvert', '--to', 'python', notebook_file], env=demo_env)
+
+        with open(python_file, 'r') as python_script:
+            # Change the working directory to the directory that contains the notebook
+            changedir = f"import os\nos.chdir(r'{os.path.dirname(python_file)}')\n"
```

Comment:
Why is this needed? The script ought to work from any directory.

Reply:
When the notebook is run, the current directory is the notebook directory, and imports from that directory work. This makes the test script do the same thing.

Reply:
> When the notebook is run, the current directory is the notebook directory

Not necessarily, the user could run it from any directory.

> and imports from that directory work

Is it really required for the current directory to be the notebook directory for the imports to work?